### PR TITLE
Proxy protocol moves tolog transport

### PR DESCRIPTION
**Pull request by bazsi**
This is now a complete LogTransport based implementation of the HAProxy protocol so it applies cleanly to any kind of LogProto implementation.

### CODE REVIEW

File: tests/light/functional_tests/source_drivers/network_source/proxyprotocol/test_pp_network.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    NUMBER_OF_MESSAGES = 2
+    _test_pp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters, '"proxied-tls"', INPUT_MESSAGES, NUMBER_OF_MESSAGES, EXPECTED_MESSAGES, TEMPLATE)
+
+
+def test_pp_network_tls_with_passphrase(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters):
+    TEMPLATE = r'"${PROXIED_SRCIP} ${PROXIED_DSTIP} ${PROXIED_SRCPORT} ${PROXIED_DSTPORT} ${PROXIED_IP_VERSION} ${MESSAGE}\n"'
+    INPUT_MESSAGES = "message 0\n"
+    EXPECTED_MESSAGES = "1.1.1.1 2.2.2.2 3333 4444 4 message 0\n"
+    NUMBER_OF_MESSAGES = 2
+    _test_pp(config, syslog_ng, syslog_ng_ctl, port_allocator, loggen, testcase_parameters, '"proxied-tls"', INPUT_MESSAGES, NUMBER_OF_MESSAGES, EXPECTED_MESSAGES, TEMPLATE, password="asdfg")
```
Comment (alltilla) [inline tests/light/functional_tests/source_drivers/network_source/proxyprotocol/test_pp_network.py:109]: We should use a different set of certs in this case:
```py
    server_key_path = copy_shared_file(testcase_parameters, "server-protected-asdfg.key")
    server_cert_path = copy_shared_file(testcase_parameters, "server-protected-asdfg.crt")
```

Same for the syslog tests.

File: lib/transport/tests/test_transport_haproxy.c [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  LogTransport *transport = log_transport_haproxy_new(&stack, LOG_TRANSPORT_INITIAL, LOG_TRANSPORT_NONE);
+
+  do
+    {
+      log_transport_aux_data_init(&aux);
+      rc = log_transport_read(transport, buf, sizeof(buf), &aux);
+    }
+  while (rc == -1 && errno == EAGAIN);
+
+  cr_assert_eq(rc == 0, params->valid,
```
Comment (alltilla) [inline lib/transport/tests/test_transport_haproxy.c:158]: I guess `%ld` is needed for the formatting of `rc`:

```
/__w/axosyslog/axosyslog/lib/transport/tests/test_transport_haproxy.c:160:38: error: format specifies type 'int' but the argument has type 'gssize' (aka 'long') [-Werror,-Wformat]
               params->proxy_header, rc, errno);
```

**Comment by alltilla** [comment]
style check fails on this:
```diff
--- a/modules/afsocket/transport-mapper-inet.c
+++ b/modules/afsocket/transport-mapper-inet.c
@@ -91,8 +91,8 @@ _setup_socket_transport(TransportMapperInet *self, LogTransportStack *stack)
 {
   log_transport_stack_add_transport(stack, LOG_TRANSPORT_SOCKET,
                                     self->super.sock_type == SOCK_DGRAM
-                                           ? log_transport_udp_socket_new(stack->fd)
-                                           : log_transport_stream_socket_new(stack->fd));
+                                    ? log_transport_udp_socket_new(stack->fd)
+                                    : log_transport_stream_socket_new(stack->fd));
   return TRUE;
 }
```

File: lib/transport/transport-stack.h [hunk 3]
```diff
@@ -61,28 +68,71 @@ log_transport_factory_free(LogTransportFactory *self)
...
+  if (self->transports[active])
+    return self->transports[active];
+
+  if (self->transport_factories[active])
+    {
+      self->transports[active] = log_transport_factory_construct_transport(self->transport_factories[active], self);
+      return self->transports[active];
+    }
+  return NULL;
+}
```
[the remark this answers is not in the document]
Comment (bazsi) [reply lib/transport/transport-stack.h]: done.

**Review by alltilla — APPROVED** [review]

**Pull request merged**